src/config.js

```diff
@@ -471,48 +471,57 @@ export default class Config {
    * throw an error if package.json was not found
    */
 
-  async readManifest(dir: string, priorityRegistry?: RegistryNames, isRoot?: boolean = false): Promise<Manifest> {
-    const manifest = await this.maybeReadManifest(dir, priorityRegistry, isRoot);
+  readManifest(dir: string, priorityRegistry?: RegistryNames, isRoot?: boolean = false): Promise<Manifest> {
+    return this.getCache(`manifest-${dir}`, async (): Promise<Manifest> => {
```

Comment:
Actually this was always a bug, but didn't surface. With this PR I actually expect manifest to change across reads, so I needed to fix this.